backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs

```diff
@@ -105,6 +118,30 @@ public async Task<List<UserSolutionsGroupedBySkillLevelDto>> GetUserChallengesIn
             .ToListAsync();
 
         return challenges;
+    }
+
+    public async Task<UserFullDto> UpdateUserAsync(EditUserDto userDto)
+    {
+        var user = await GetUserByExpressionAsync(user => user.Id == userDto.Id)
+            ?? throw new ArgumentNullException(nameof(userDto));
+
+        user.TotalScore += await GetRewardFromChallenge(userDto.CompletedChallengeId);
+        user.Reputation = user.TotalScore / REPUTATION_DIVIDER;
+
+        _context.Users.Update(user);
+
+        await _context.SaveChangesAsync();
+
+        return _mapper.Map<UserFullDto>(user);
+    }
+
+    private async Task<int> GetRewardFromChallenge(long challengeId)
+    {
+        var challenge = await _context.Challenges
+            .Include(challenge => challenge.Level)
+            .SingleOrDefaultAsync(challenge => challenge.Id == challengeId);

```

Comment:
you can use Select for loading only what you need (not a full entity with all its dependencies)